File: GO/euler007.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(solve())
}

func solve() int {
	LIMIT := 10001
	lst := []int{}
	i := 1

	for len(lst) < LIMIT {
		i++
		if is_prime(i) {
			lst = append(lst, i)
		}
	}
	return lst[len(lst)-1]
}

func is_prime(p int) bool {
	for i := 2; i <= int(math.Sqrt(float64(p))); i++ {
		if p%i == 0 {
			return false
		}
	}
	return true
}

File: C#/euler007.cs
using System;

namespace ProjectEuler {
    class euler007 {
        static void Main() {
            Console.WriteLine(Solve());
        }

        static int Solve() {
            int counter = 0;

            for (int i = 2; ; i++) {
                if (is_prime(i)) {
                    counter++;
                }
                if (counter == 10001) {
                    return i;
                }
            }
        }

        static bool is_prime(int n) {
            int end = Convert.ToInt32(Math.Sqrt(Convert.ToDouble(n)));

            for (int i = 2; i <= end; i++) {
                if (n%i == 0) {
                    return false;
                }
            }
            return true;
        }
    }
}